### PR TITLE
Add description-manager reusable workflow (replaces bump_dev_version_tag_branch)

### PR DESCRIPTION
**Pull request by Felixmil**
## Summary

Replaces `bump_dev_version_tag_branch.yaml` with a new reusable workflow `description_manager.yaml`.

The old workflow is kept as a thin shim that delegates to the new one and emits a deprecation warning, so **existing callers keep working unchanged**.

## Why

**No more GitHub-App token, no more secret.** The old workflow required `app-id` input + `private-key` secret to mint a token via `actions/create-github-app-token`. The new workflow uses the default `GITHUB_TOKEN` plus `stefanzweifel/git-auto-commit-action@v7`. The shim keeps the old inputs/secret in its signature (now ignored) so existing callers don't break.

```yaml
# Before
uses: Open-Systems-Pharmacology/Workflows/.github/workflows/bump_dev_version_tag_branch.yaml@<ref>
with:
  app-id: ${{ vars.APP_ID }}
secrets:
  private-key: ${{ secrets.APP_PRIVATE_KEY }}

# After
uses: Open-Systems-Pharmacology/Workflows/.github/workflows/description_manager.yaml@<ref>
permissions:
  contents: write
```

Other improvements:

- Bash logic extracted to `.github/scripts/manage_description.sh`, covered by 11 bats tests run in CI (caught a real BSD-vs-GNU awk portability bug in the original).
- Push-only event gate (skips `workflow_dispatch` / `schedule` / `pull_request`).
- Per-ref concurrency so different branches don't block each other.
- Rich step summary with old → new version, Remotes diff, linked commit & tag.

## Step summary examples

**Dev bump** — https://github.com/Felixmil/Workflows-test-pkg/actions/runs/25522748964

> ## Update DESCRIPTION
> 
> **Version:** `0.1.1.9002` → `0.1.1.9003` (dev bump)
> 
> ### Remotes
> 
> _Remotes already in expected state; no edit applied._
> 
> ### Pushed
> 
> - **Commit:** [`4eb559f`](https://github.com/Felixmil/Workflows-test-pkg/commit/4eb559f5af06f4cd47d0929b17c390a4fe382284)
> - **Tag:** _(none — dev bumps are not tagged)_

**Release** — https://github.com/Felixmil/Workflows-test-pkg/actions/runs/25523185106

> ## Update DESCRIPTION
> 
> **Version:** `0.2.0` (release, no bump)
> 
> ### Remotes
> 
> Added `@*release` (release version).
> 
> <details><summary>Remotes block before & after</summary>
> 
> **Before:**
> 
> ```
> Remotes:
>     tidyverse/dplyr
> ```
> 
> **After:**
> 
> ```
> Remotes:
>     tidyverse/dplyr@*release
> ```
> 
> </details>
> 
> ### Pushed
> 
> - **Commit:** [`5d629ad`](https://github.com/Felixmil/Workflows-test-pkg/commit/5d629aded032fabe64de26d6c6ce7350ea291dfb)
> - **Tag:** [`v0.2.0`](https://github.com/Felixmil/Workflows-test-pkg/releases/tag/v0.2.0)

## Summary by CodeRabbit

* **Chores**
  * Refactored version management workflow into a new reusable component for automated DESCRIPTION file updates, consolidating version bumping and dependency pinning. Previous workflow implementation deprecated in favor of the new reusable design.
  * Added comprehensive automated test suite validating version updates and dependency management to ensure functional reliability.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Open-Systems-Pharmacology/Workflows/pull/225)

### CODE REVIEW

File: .github/workflows/test_description_manager.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  bats:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v6
+
+      - name: Install bats
+        run: sudo apt-get update && sudo apt-get install -y bats
+
+      - name: Run bats tests
+        run: bats .github/scripts/tests/
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply .github/workflows/test_description_manager.yaml]: addressed

File: .github/scripts/tests/manage_description.bats
Comment (Yuri05) [inline .github/scripts/tests/manage_description.bats:1]: @Felixmil What is this manage_description.bats and where is it used?

File: .github/workflows/description_manager.yaml [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    if: github.event_name == 'push'
+    concurrency: #prevent concurrent runs on the same ref
+      group: ${{ github.workflow }}-${{ github.ref }}
+    runs-on: ubuntu-latest
+    outputs:
+      run_next_job: ${{ steps.set_output.outputs.run_next_job }}
+    steps:
+      - uses: actions/checkout@v6
+        with:
+          fetch-depth: 0
```
Comment (Yuri05) [inline .github/workflows/description_manager.yaml]: Why do we need the whole history here?

File: .github/workflows/description_manager.yaml [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            NAMESPACE
+            DESCRIPTION
+            NEWS.md
+            .Rbuildignore
+            **.R
+            tests/**
+            vignettes/**
+            inst/**
+            man/**
+
```
Comment (Yuri05) [inline .github/workflows/description_manager.yaml:65]: uff, dangerous. what if I change any of the required files not in this list?
e.g. under **shared/** in the rSharp package, or a picture which is referenced in README.md or ...

File: .github/workflows/description_manager.yaml [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    concurrency: #prevent concurrent runs on the same ref
+      group: ${{ github.workflow }}-${{ github.ref }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout caller repo
+        uses: actions/checkout@v6
+        with:
+          fetch-depth: 0
```
Comment (Yuri05) [inline .github/workflows/description_manager.yaml]: why the whole history?

File: .github/scripts/tests/manage_description.bats
Comment (Felixmil) [reply .github/scripts/tests/manage_description.bats:1]: It's test for the manage-description script. Ran by gha in .github/workflows/test_description_manager.yaml

**Comment by Yuri05** [comment]
> **No more GitHub-App token, no more secret.** The old workflow required `app-id` input + `private-key` secret to mint a token via `actions/create-github-app-token`. The new workflow uses the default `GITHUB_TOKEN` plus `stefanzweifel/git-auto-commit-action@v7`. The shim keeps the old inputs/secret in its signature (now ignored) so existing callers don't break.

That's very nice!

File: .github/workflows/description_manager.yaml (hunk 3 above)
Comment (Felixmil) [reply .github/workflows/description_manager.yaml:65]: It was exactly like this in original bump_dev_version_tag_branch.yaml

But we can broaden the list !

File: .github/workflows/description_manager.yaml (hunk 4 above)
Comment (Felixmil) [reply .github/workflows/description_manager.yaml]: Was like this in bump_dev_version_tag_branch.yaml
We can try to remove it

File: .github/scripts/tests/manage_description.bats
Comment (Yuri05) [reply .github/scripts/tests/manage_description.bats:1]: OK, thx

File: .github/workflows/description_manager.yaml (hunk 3 above)
Comment (Yuri05) [reply .github/workflows/description_manager.yaml:65]: Can we generically turn this list into "_Everything except what is included in **.Rbuildignore**_" ?

File: .github/workflows/description_manager.yaml (hunk 2 above)
Comment (Yuri05) [reply .github/workflows/description_manager.yaml]: 2 should be sufficient (compare https://github.com/tj-actions/changed-files#usage-)

File: .github/workflows/description_manager.yaml (hunk 3 above)
Comment (Yuri05) [reply .github/workflows/description_manager.yaml:65]: > Can we generically turn this list into "_Everything except what is included in **.Rbuildignore**_" ?

~Seems that the used action does not allow negative filters. As a compromise:~
* ~get the list of all **top level** file/folder names from the repo~
* ~exclude those top level names which are included into **.Rbuildignore**~
~E.g. if **.Rbuildignore** contains `.github` and `tests/dev`: the final list would include every top level name except `.github`~

Comment (Yuri05) [reply .github/workflows/description_manager.yaml:65]: no, actually it supports negative filters, with `!`

File: .github/workflows/description_manager.yaml (hunk 2 above)
Comment (Felixmil) [reply .github/workflows/description_manager.yaml]: Reduced to `fetch-depth: 2` in acf3d92, per the [tj-actions/changed-files docs](https://github.com/tj-actions/changed-files#usage-).

File: .github/workflows/description_manager.yaml (hunk 4 above)
Comment (Felixmil) [reply .github/workflows/description_manager.yaml]: Same — reduced to `fetch-depth: 2` in acf3d92.

File: .github/workflows/description_manager.yaml [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+            done
+          } >> "$GITHUB_STEP_SUMMARY"
+
+          echo "run_next_job=true" >> $GITHUB_OUTPUT
+
+      - name: No changes
+        if: steps.changes.outputs.any_changed != 'true'
+        run: |
+          echo "::notice title=No changes::No functional changes detected, DESCRIPTION will not be updated"
+          echo "run_next_job=false" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply .github/workflows/description_manager.yaml]: Quoted in acf3d92.

File: .github/workflows/description_manager.yaml (hunk 3 above)
Comment (Felixmil) [reply .github/workflows/description_manager.yaml:65]: Update on this thread — after running the `.Rbuildignore` translator on 12 OSP R packages, we found that real-world `.Rbuildignore` files contain a fair amount of regex idioms that don't translate cleanly to globs (typos like `\\cache`, character classes, mid-pattern `*`, etc.). Rather than maintain a translator with edge cases, we pivoted in 4f7bedb:

- The workflow now exposes a `tracked-files` input with a sensible default list, broadened from the original to address your concern: it now includes `shared/**`, `README*.png` (and other README assets), `data/**`, `data-raw/**`, `src/**` in addition to the original `R/`, `tests/`, `vignettes/`, `inst/`, `man/`, README/NAMESPACE/NEWS/DESCRIPTION.
- Callers that need something different can override per-repo:

```yaml
uses: Open-Systems-Pharmacology/Workflows/.github/workflows/description_manager.yaml@main
with:
  tracked-files: |
    R/**
    custom/**
```

The translator script and its bats tests have been removed.

Comment (Felixmil) [reply .github/workflows/description_manager.yaml:65]: Re: your `.Rbuildignore` + negative filter idea — I prototyped it (a script that translated `.Rbuildignore` regex into glob `!` patterns / `files_ignore` entries) and ran it against 12 OSP R packages to see how it would behave. The results made us drop the approach. Three reasons:

**1. `.Rbuildignore` uses Perl regex, not globs.** The action expects globs, so every line needs translation. That sounds simple but isn't:

- `^.*\.Rproj$` is universal across OSP packages and means "any `.Rproj` file" — translates to `**/*.Rproj`. Fine.
- `^[.]?air[.]toml$` (character class for optional dot) needs to be expanded to two patterns: `air.toml` and `.air.toml`.
- `^paper.*$`, `publication/*`, `^tests/dev/*\.R$` — `*` in different positions, each needing different glob handling.
- `(^appveyor)(.*)(.yml$)` — capture groups, no glob equivalent.
- Several typos in real OSP `.Rbuildignore` files: `^\\cache$` (meant `^cache$` — `\\c` is just literal `c`), `tests\\^spelling` (`\\^` is a literal `^`, matches nothing). The regex compiles but doesn't do what the author thought. A translator either silently drops these (bug-for-bug fidelity) or warns and skips them (correct but noisy in CI logs).

**2. The downside of getting it wrong is asymmetric.** If we under-translate an ignore pattern, we *over-trigger* the workflow — every push that touches an `.Rproj` would needlessly bump DESCRIPTION. A predefined allow-list fails the other way: it under-triggers, which is more annoying but never wrong (you can always re-run manually).

**3. Maintenance cost.** The translator was ~130 lines of bash with bats tests covering ~15 edge cases, and would have grown as we hit new `.Rbuildignore` quirks across the org. Compared to the current ~15-line YAML default (which a caller can override per-repo), it's a lot of moving parts for a workflow that runs once per merge.

So we kept it simple: predefined sensible default, override via input. If a caller has an unusual layout (e.g. `shared/` in rSharp), they pass a custom `tracked-files` once and we never touch their `.Rbuildignore`.

I kept the prototype locally if you ever want to look at the per-package translation report — happy to share it.

Comment (Yuri05) [reply .github/workflows/description_manager.yaml:65]: Sounds as a good compromise and let's keep it like this for now. 
The effort you described above for the "universal" Rbuildignore-translation seems indeed way too high.

Comment (Yuri05) [reply .github/workflows/description_manager.yaml:65]: Additionally I would add a description of this procedure in our [dev docs](https://dev.open-systems-pharmacology.org/r-development-resources/best-practices-r). 
This is something which will be easily forgotten otherwise.

**Review by Yuri05 — DISMISSED** [review]
OK to merge IMO (once the merge conflicts are resolved)

File: .github/workflows/description_manager.yaml (hunk 3 above)
Comment (Felixmil) [reply .github/workflows/description_manager.yaml:65]: I will add a "Versionning" section there

**Comment by Felixmil** [comment]
New workflow now uses `vars.GHA_DEFAULT_RUNNER_*` too.
